kt/godot-library/src/main/kotlin/godot/annotation/propertyHintAnnotations.kt

```diff
@@ -0,0 +1,265 @@
+@Retention(AnnotationRetention.RUNTIME)
+annotation class ExpRange(val start: Float, val end: Float, val step: Float = -1f)
+
+/**
+ * Can only be used on Enum properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class EnumTypeHint
+
+/**
+ * Can only be used on Float and Double properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class ExpEasing(val attenuation: Boolean = false, val inout: Boolean = true)
+
+/**
+ * Can only be used on Float and Double properties!
+ */
+//@Target(AnnotationTarget.PROPERTY)
+//@Retention(AnnotationRetention.RUNTIME)
+//annotation class Lenght(val lenght: Int = -1)
+
+/**
+ * Flag Property hint supporting enums
+ *
+ * Can only be used on Set<Enum> or MutableSet<Enum> properties!
+ *
+ * **Note:** The used enum can not have more than 32 Entries!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class EnumFlag
+
+/**
+ * Flag Property hint supporting Int's
+ *
+ * Can only be used on Int properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class IntFlag(vararg val names: String)
+
+/**
+ * Can only be used on Int properties!
+ */
+//@Target(AnnotationTarget.PROPERTY)
+//@Retention(AnnotationRetention.RUNTIME)
+//annotation class Layers2DRender
+
+/**
+ * Can only be used on Int properties!
+ */
+//@Target(AnnotationTarget.PROPERTY)
+//@Retention(AnnotationRetention.RUNTIME)
+//annotation class Layers2DPhysics
+
+/**
+ * Can only be used on Int properties!
+ */
+//@Target(AnnotationTarget.PROPERTY)
+//@Retention(AnnotationRetention.RUNTIME)
+//annotation class Layers3DRender
+
+/**
+ * Can only be used on Int properties!
+ */
+//@Target(AnnotationTarget.PROPERTY)
+//@Retention(AnnotationRetention.RUNTIME)
+//annotation class Layers3DPhysics
+
+/**
+ * Can only be used on String properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class File(vararg val extensions: String = [], val global: Boolean = false)
+
+/**
+ * Can only be used on String properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class Dir(val global: Boolean = false)
+
+/**
+ * Can only be used on String properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class MultilineText
+
+/**
+ * Can only be used on String properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class PlaceHolderText
+
+/**
+ * Can only be used on Color properties!
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class ColorNoAlpha
+
+/**
+ * Can only be used on Image properties!
+ */
+//@Target(AnnotationTarget.PROPERTY)
+//@Retention(AnnotationRetention.RUNTIME)
+//annotation class ImageCompressLossy
```

Comment:
Is this for future or should it be removed ?

Reply:
These are for future implementation. We did not add those to the kotlin-native project because we were not sure how they work and though we should only add them once we know how they work. So i did the same here.
Once we have spare time we can add these. I thought they just are not that important atm.